### DISCUSSION TITLE
Constant connection losses

### DISCUSSION THREAD

**Discussion by dz46** [NightscoutFoundation/xDrip#3643] [Q&A]
He, I’m currently running version f96d4e5-2024.08.07 on a Giagset E940-2878-03 on Android 12 Kernel 4.19.191.

Since the switch to G7 I have constantly (sometimes 60%+ of the day or more) no “new” data in xdrip. It also takes me a while to notice, that there a no new data since it happens so often and sometimes it fixes itself after opening the app in foreground and then locking the phone again.

Most of the times it helps, to restart data collector manually (which solves ~ 80%) of the occurences. In the other instances, I unpair the transmitter and change the transmitter id to somthing in xdrip and back, to so it can re-pair again, which can also takes sometime up to 30 min.

I have already gone through the normal steps:
- Checked that xDrip battery optimization is disbaled?
- Checked that  Bluetooth battery optimization is off?
- hanged all setting according to https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html and https://navid200.github.io/xDrip/docs/Connectivity-troubleshoot.html

Seems not be a general BT issue since, the Dana-i also connected, has 0 issue with BT (or the app being killed) - it also worked with G6 like charm.

Any other ideas to tackle this? (Besides buying new phone and hoping for the best)

![Screenshot_20240826-105105|100](https://github.com/user-attachments/assets/117705d1-1c04-4c04-9178-18907bd9b7e5)
![Screenshot_20240826-105042](https://github.com/user-attachments/assets/ac736dc6-d2a9-4243-9ecc-478376594d9e)

**Comment by Navid200** [comment]
How exactly is Dana connected?
Do you have any reference showing operation at the same time of your Dana software and Dexcom app?

**Reply by dz46** [reply]
The Dana is running via AndroidAPS (on the same mobile), but there is no connection from Dana to G7 directly

**Reply by Navid200** [reply]
Then, please read your statement (captured in the attached image) and explain what it means.   
<br/> 
  
![Screenshot 2024-08-26 083230](https://github.com/user-attachments/assets/21057842-39e6-4c96-ad27-b00421175be4)

**Reply by Navid200** [reply]
May be I see what you are saying.
  
Does Dana or AAPS only communicate for a few seconds only once every 5 minutes like G7 does?

**Reply by Navid200** [reply]
xDrip has to wake up and scan at the right time.  If it scans continuously, the phone battery will drain rapidly.  It will also interfere with the recent Android limitations on Bluetooth scan time.  So, we have to limit how long xDrip can scan.  But, you have some control over those by the avoid scanning and minimize scanning settings.  

The Bluetooth disconnect issues you describe are common when either xDrip battery or Bluetooth battery optimization has not been disabled.
On some phones, even if you disable xDrip battery optimization, the manufacturer version of Android running on the phone still interferes.  Xiaomi and Samsung are two examples.

Is Dexcom app compatible with your phone to try?  If Dexcom app runs fine and xDrip does not, we should fix xDrip.

**Reply by dz46** [reply]
AndroidAPS is actively connection every couple min, when it sends commands to the pump. Changing baseal-rate or giving bolus. 

I double checked, battery optimization is off for xdrip & BT. 

The phone is a Gigaset and it seems the only change they did from the vanilla version is, changing the boot logo..

The original Dexcom App is not campatible with my phone unfortunalety. Maybe some own build version would work.

**Reply by Navid200** [reply]
There is someone who releases a modified version of the Dexcom app that  installs on all phones.  Have you tried that?

**Reply by dz46** [reply]
I tried the DiaKEM G7 v1.6.1.4537 - but with same result.. was woken up at least 3x because of connection loss. 
Fixed itself in the dexcom app, but just opening in foreground and waiting.   (Also looks simlar to [/discussions/3611](/NightscoutFoundation/xDrip/discussions/3611))

**Reply by Navid200** [reply]
So, it's not specific to xDrip, right?

No, Pixel 7 is not a problem at all.  The other discussion never concluded.  But, it showed very low voltages.  It was more likely related to defective sensors.

**Comment by dz46** [comment]
No it's nit specific to xDrip, since the same behavior with  Dexcom app..  so I guess i need another phone for that 🙄

**Comment by jamorham** [comment]
Please enable debugging for OB1 collector:
https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector

Wait some time for data to be gathered and next time you notice a problem, please us the 3 dot menu on the top right of xDrip home screen and select the Event log and then use the Log Upload feature and then come back here and post the date and time you made the upload and any identifier you used and I will examine the logs to see if there is any clue what is wrong.

**Reply by dz46** [reply]
Hey, I just uploaded the logs they contain debug data, until I changed to the dexcom app yesterday.

**Reply by Navid200** [reply]
So, you hadn't scanned the QR code when you were using xDrip yesterday, right?
If that's the case, that's not what he asked for.  
He needs to see the detail logs, not regular logs, when xDrip is active and is not connecting.

**Reply by dz46** [reply]
I mwean I can see the debug logs on the 26.08 19:00CEST  - I scanned the QR code before you asked me, because I thought it meigth be useful later..

**Reply by Navid200** [reply]
Great then

**Comment by jamorham** [comment]
Also please describe in more detail what happens when it is not working. Do you get missed readings or do you get it unpairing?

**Reply by dz46** [reply]
So I get no new data from the sensor.. Dexcom calls it signalloss. Putting the app into foreground & restarting data collector, will get me data at some point in time, although can take couple of minutes, until it workes again..

**Comment by jgslade** [comment]
I have also been getting connection issues between my G7 and my Pixel 8 Pro, though when I use the Dexcom App connecting to Xdrip with the companion app feature I get great connections.  Connection problems will radomly start some time between days 1-4, but once they start I'll get like 1 or 2 connections per hour using Xdrip, but when I switch to companion mode in Xdrip and connect to the sensor in the Dexcom app I don't have connection issues.

I saw Jamorham's message up above about enabling debugging for the ob1 collector, I have done that and will post logs when I start my next sensor, I couldn't get Xdrip to reconnect to my current one even with disabling Bluetooth for the Dexcom app and force closing it. 

I don't have any issues connecting to other bluetooth devices include my omnipod.

My setup:

Pixel 8 Pro
Dexcom G7
Omnipod Dash
AAPS

**Reply by Navid200** [reply]
@jgslade It doesn't help you or us if you find a thread with the only common parameter with yours being G7 and post in it instead of opening a new thread.
Pixel is a solid phone and night and day different than any other phone.
